### PR TITLE
Add compact view of timer records

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds special formatting for timer records when printed as part of the compacted log. It will show the target element ID, the process definition key, either that it is a start event or its element instance key and its process instance key, when it's due (omitting the date part if its the same as the current date), and how many repetitions it has left.

Samples:

```
C TIMER      CREATE      - #15->#14  -1 -  @"timer"[K07] in <process "K01"[K03]> due T10:17:47.856
E TIMER      CREATED     - #6->#3    K3 -  @"timer"[-1]  in <process "K1"[-1]> due T10:18:16.072
E TIMER      CREATED     - #6->#3    K3 -  @"timer"[-1]  in <process "K1"[-1]> due T10:13:11.122 2 reps
```

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: test-util/src/main/java/io/zeebe/test/util/record/CompactRecordLogger.java [hunk 1]
```diff
@@ -455,6 +459,35 @@ private StringBuilder summarizeRejection(final Record<?> record) {
...
+    final var builder = new StringBuilder();
+    final var dueTime = Instant.ofEpochMilli(value.getDueDate()).atZone(ZoneId.systemDefault());
+
+    builder.append("@").append(formatId(value.getTargetElementId()));
+
+    if (value.getElementInstanceKey() > 0) {
+      builder.append("[").append(shortenKey(value.getElementInstanceKey())).append("]");
+    } else {
+      builder.append(" (start event)");
+    }
```
Comment (saig0) [inline test-util/src/main/java/io/zeebe/test/util/record/CompactRecordLogger.java:473]: Following DRY, we should use `summarizeElementInformation()` instead. 
For timer start events, the element instance key could be `-1` or `?`.

File: test-util/src/main/java/io/zeebe/test/util/record/CompactRecordLogger.java [hunk 2]
```diff
@@ -455,6 +459,35 @@ private StringBuilder summarizeRejection(final Record<?> record) {
...
+
+    builder
+        .append(" in <process [")
+        .append(shortenKey(value.getProcessDefinitionKey()))
+        .append("]>");
+
+    if (value.getProcessInstanceKey() > 0) {
+      builder.append(" instance <[").append(shortenKey(value.getProcessInstanceKey())).append("]>");
+    }
```
Comment (saig0) [inline test-util/src/main/java/io/zeebe/test/util/record/CompactRecordLogger.java:482]: Following DRY, we should use `summarizeProcessInformation()` instead.

**Review by saig0 — CHANGES_REQUESTED** [review]
@npepinpe thanks :cookie: 

I've suggestions to reduce duplication and align with the format. Please have a look.

File: test-util/src/main/java/io/zeebe/test/util/record/CompactRecordLogger.java (hunk 2 above)
Comment (npepinpe) [reply test-util/src/main/java/io/zeebe/test/util/record/CompactRecordLogger.java:482]: Here we don't have the bpmnElementId but rather the process definition key. We could add an overload for this specific case I suppose, which would show:

- ` in <process K03[K05]> ` for normal catch events
- ` in <process K03[?]> ` for start events

Of course I can always format the key and pass that as the `bpmnElementId`, which will do the same, but may look strange in the future if we change how we format IDs :sweat_smile: Actually, thinking about it, it's probably fine for now to do that.

**Comment by npepinpe** [comment]
I've updated based on your suggestions, and update the PR comment with the new "look".

**Review by saig0 — APPROVED** [review]
:tada:

**Pull request merged**